hyperspy/tests/io/test_io.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+# Copyright 2007-2016 The HyperSpy developers
+#
+# This file is part of  HyperSpy.
+#
+#  HyperSpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+#  HyperSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with  HyperSpy.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import numpy as np
+import pytest
+from unittest.mock import patch
+
+from hyperspy.signals import Signal1D
+from hyperspy.io import load
+
+dirpath = os.path.dirname(__file__)
+filename = 'test_io_overwriting.hspy'
+fullfilename = os.path.join(dirpath, filename)
+
+
+class TestIOOverwriting:
+
+    def setup_method(self, method):
+        self.s = Signal1D(np.arange(10))
+        self.new_s = Signal1D(np.ones(5))
+        # make sure we start from a clean state
+        self._clean_file()
+        self.s.save(fullfilename)
+
+    def _clean_file(self):
+        if os.path.exists(fullfilename):
+            os.remove(fullfilename)
+
+    def _check_file_is_written(self, signal, filename):
```

Comment:
Checking in this way is a bit weak and may get broken for other reasons. What about using the hashlib instead to computer e.g. sha256 or simply verify the the modification date has changed in respect to the original?